processor/src/host/mod.rs

```diff
@@ -111,16 +124,30 @@ pub trait AsyncHost: BaseHost {
 
     /// Returns MAST forest corresponding to the specified digest, or None if the MAST forest for
     /// this digest could not be found in this [AsyncHost].
-    fn get_mast_forest(
-        &self,
-        node_digest: &Word,
-    ) -> impl Future<Output = Option<Arc<MastForest>>> + Send;
+    fn get_mast_forest(&self, node_digest: &Word) -> Option<Arc<MastForest>>;
 
     /// Handles the event emitted from the VM and provides advice mutations to be applied to
     /// the advice provider.
     fn on_event(
         &mut self,
         process: &ProcessState<'_>,
         event_id: u32,
-    ) -> impl Future<Output = Result<Vec<AdviceMutation>, EventError>> + Send;
+    ) -> impl AsyncHostFuture<Result<Vec<AdviceMutation>, EventError>>;
 }
+
+/// Alias for a `Future`
+///
+/// Depending on the the target being `wasm32-unknown-unknown` the
+/// trait bounds are _without_ `+ Send`, otherwise with `+ Send` to
+/// ensure functionality for mutli threaded executor.
```

Comment:
This comment is outdated - we not use `std` feature. But as discussed at some point offline, is it correct to depend on `std`? Basically AFAIU if any dependency depends on `std`, then the final binary will be compiled with `std` (due to cargo features being additive). This was my understanding why using `#[cfg(target_family = "wasm")]` as suggested by @bitwalker in https://github.com/0xMiden/miden-vm/pull/2033#discussion_r2240040800 as preferred